### ISSUE TITLE
Water disappears when zooming in closely

### ISSUE THREAD

**Issue by njohnston** [osmandapp/Osmand#2805]
(As requested by @xmd5a2 in #2607 I'm logging a separate issue for this.)

When zooming in on [Lundy, a small island off the coast of England](http://www.openstreetmap.org/#map=13/51.1896/-4.6539), water sometimes disappears.

At zoom 12, the island is rendered correctly:

![osmand_water_issue_ok_small](https://cloud.githubusercontent.com/assets/638458/16422782/8c124f50-3d51-11e6-88df-db548d4572fc.jpg)

At zoom 13, the water disappears:

![osmand_water_issue_broken_small](https://cloud.githubusercontent.com/assets/638458/16422768/83e73976-3d51-11e6-8c4e-1c50ef67e897.jpg)

logcat shows:

```
I net.osmand: MapRenderRepositories RENDER MAP: new request RotatedTileBox [lat=51.179412841796875, lon=-4.666409969329834, rotate=0.0, density=3.0, zoom=13, mapDensity=3.0, zoomAnimation=0.0, zoomFloatPart=0.0, cx=900, cy=1386, pixWidth=1800, pixHeight=2772]
I net.osmand:native: Route objects 182
I net.osmand:native: Islands 33, closed water 2, coastline touches screen 0
```

The water is also missing at zoom 14, but returns at zoom 15:

![osmand_water_issue_ok_higher_zoom_small](https://cloud.githubusercontent.com/assets/638458/16422891/e5376746-3d51-11e6-93d2-5c852ba79b20.jpg)

At zoom 15, logcat includes:

```
I net.osmand: MapRenderRepositories RENDER MAP: new request RotatedTileBox [lat=51.179412841796875, lon=-4.666409969329834, rotate=0.0, density=3.0, zoom=15, mapDensity=3.0, zoomAnimation=0.0, zoomFloatPart=0.0, cx=900, cy=1386, pixWidth=1800, pixHeight=2772]
I net.osmand:native: Route objects 284
I net.osmand:native: Islands 22, closed water 5, coastline touches screen 1
```

I'm using OsmAnd+ 2.3.5 on a Samsung Galaxy S5 with the offline map of England ("United Kingdom England") from 1st June. (There is no July map update for England yet.)

**Comment by xmd5a2** [comment]
Note: OpenGL rendering has no such a problem.

**Comment by njohnston** [comment]
@xmd5a2: I enabled the development plugin and "Use OpenGL rendering" is ticked. Is there any to find out if OpenGL is definitely in use?

**Comment by xmd5a2** [comment]
@njohnston this will work only with http://download.osmand.net/latest-night-build/OsmAnd-qt-arm-nightly.apk

**Comment by njohnston** [comment]
@xmd5a2: I downloaded the APK and tried to install it with `adb install` but got this error: INSTALL_FAILED_UPDATE_INCOMPATIBLE. Is this because I've got OsmAnd+ installed from the Play Store? Do any of the other APKs at http://download.osmand.net/latest-night-build/ have OpenGL support?

**Comment by xmd5a2** [comment]
@njohnston Yes, qt-arm-nightly replaces osmand+. No, only this package have OpenGL support. You can turn on development plugin and download nightly using main menu - Builds.

**Comment by njohnston** [comment]
@xmd5a2: that's a shame. I don't want to lose the stable OsmAnd+. It's too important an app for me, especially when travelling. Would you be able to enable OpenGL on the nightly build of the free version? I'd like to help test it.

**Comment by xmd5a2** [comment]
@njohnston There is no need to test it. There are many known issues and most important is rendering speed which is under development now.

**Comment by vshcherb** [comment]
Just to correct @xmd5a2, the rendering speed is faster on new devices in qt-arm build, the layer rendering is slower and on most of the zooms it is ok, though for 5-7 zoom it is catastrophic slower. Also a known issue rotation of the screen and road only maps (not supported at all)

**Comment by njohnston** [comment]
@xmd5a2: OK. Out of interest, will using OpenGL allow 3D/2.5D rendering for navigation?

**Comment by xmd5a2** [comment]
@njohnston yes it will

**Comment by vshcherb** [comment]
You can already give a try to it http://download.osmand.net/latest-night-build/ (sample apk and qt apk)

**Comment by njohnston** [comment]
@vshcherb: I downloaded today's sample build (OsmAnd-qt-sample-armv7.apk) and installed it (on a Samsung Galaxy S5 running Android 6.0.1) with `adb install`. It force closes every time I run it:

```
net.osmand: MapRenderingTypes Unexpected error
net.osmand: java.io.FileNotFoundException: /storage/emulated/0/osmand/poi_types.xml:
open failed: EACCES (Permission denied)
```

I've seen others reporting this but I'm not sure where. If I assign the app storage permission, it no longer crashes on start-up. However, the app doesn't display any maps:

![osmand_sample_small](https://cloud.githubusercontent.com/assets/638458/17139658/4da4d7e0-533d-11e6-859f-db395415e563.jpg)

**Comment by vshcherb** [comment]
yes we have an issue already for sample apk. Will be fixed soon. 
Also please keep in mind that for now sample requires maps from OsmAnd itself (so osmand should be also installed and use shared_storage)

**Comment by annasoloviova** [comment]
Same in Petrozavodsk, Russia.

![scr1](https://cloud.githubusercontent.com/assets/12711814/17366993/c21d19b6-5996-11e6-8497-d184b46f2f18.png)
![scr2](https://cloud.githubusercontent.com/assets/12711814/17366994/c21da5c0-5996-11e6-98e6-e4ebba899f9e.png)

**Comment by annasoloviova** [comment]
Also happens in France. http://osmand.net/go?lat=43.324505&lon=3.5483599&z=16 on 100m zoom and closer.

**Comment by sonora** [comment]
Let me close this one, no actionable task here for now.

**Comment by xmd5a2** [comment]
The bug is not resolved and very annoying

**Comment by sonora** [comment]
Okay, maybe I misunderstood, but I thought we are not fixing this in the old engine any more ?

**Comment by xmd5a2** [comment]
Related to https://github.com/osmandapp/Osmand/issues/3596
